### ISSUE TITLE
Client cannot connect on Linux (Fedora 26 kernel 4.13.8-200.fc26.x86_64)

### ISSUE THREAD

**Issue by broxen** [Segs/Segs#62]
The CoX client crashes immediately on Linux and shows two dialog boxes with error messages (below). Maybe I'm setting up the database incorrectly? Or the server config?

I'm trying to connect to my local IP, not a loopback address.

It works on Windows, with the same setup.

Let me know what other information you may need.

**uname -a**
`Linux zelda.hyrule 4.13.8-200.fc26.x86_64 #1 SMP Wed Oct 18 15:42:58 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`

**Server Log (output.log)**
```
2017-10-20 02:42:12.671@@11835@LM_ERROR@main
2017-10-20 02:42:12.672@@11835@LM_WARNING@(11835|139669139534976) Reading game data
2017-10-20 02:42:12.672@@11835@LM_WARNING@(11835|139669139534976) Reading Costumes ... 2017-10-20 02:42:12.672@@11835@LM_WARNING@(11835|139669139534976) Couldn't load Costume!!
2017-10-20 02:42:12.672@@11835@LM_WARNING@(11835|139669139534976) Reading Supergroup colors ... 2017-10-20 02:42:12.672@@11835@LM_WARNING@(11835|139669139534976) Couldn't load Supergroup colors!!
2017-10-20 02:42:12.672@@11835@LM_WARNING@(11835|139669139534976) All game data read
2017-10-20 02:42:12.672@@11835@LM_WARNING@(11835|139669139534976) Filling hashes .. 2017-10-20 02:42:12.672@@11835@LM_WARNING@Hashes filled
```

**Client Error Log**
```
fixme:winediag:start_process Wine Staging 2.18 is a testing version containing experimental patches.
fixme:winediag:start_process Please mention your exact version when filing bug reports on winehq.org.
Running City Of Heroes
Error opening config file (N:/revisions/gimmecfg.txt)                          
Will proceed without source control

You probably have been disconnected from N:, double click on My Computer
and then double click on N: to reconnect it.
project segs
fullscreen 0
Caching relevant folders.. done                                       (0.00)
Error loading required game data files.  (trick load error)

Loading texture headers..                                             (0.00)
```

**Dialog Box 1**
Error loading required game data files.  (trick load error)

**Dialog Box 2**
Failed to load texture library.
It is likely that City of Heroes is not properly installed.  Re-installing or running CohUpdater may fix this.

**Comment by nemerle** [comment]
In order:
We need to fix the server messages to report cleaner errors on startup: #64
The client's report of ```Error opening config file (N:/revisions/gimmecfg.txt)``` suggests it's in developer mode, that happens if:
* it cannot find ./piggs directory
* or it finds a ./data directory

**Comment by broxen** [comment]
I can confirm that the ./piggs directly exists and is the same as the windows install. I tried both a fresh install of the CoX SEGS client and tried copying over from my working windows installation.

I have no ./data directory in the client folder.

My command line arguments are `-project segs -fullscreen 0` trying without fullscreen has no impact.

**Comment by nemerle** [comment]
Are You starting it up from commandline, or the desktop-created icon ?

**Comment by broxen** [comment]
From the terminal:
```
env WINEPREFIX="/home/broxen/.wine32" wine /home/broxen/.wine32/dosdevices/c:/Program\ Files/CoX/CoX.exe -project segs -fullscreen 0 -screen 1280 720
```

with or without the fullscreen and screen switches.

**Comment by nemerle** [comment]
could You try doing:
```bash
cd /home/broxen/.wine32/dosdevices/c:/Program\ Files/CoX/
wine CoX.exe -project segs -fullscreen 0 -screen 1280 720
```
and also checking with ```regedit``` if the HKEY_CURRENT_USER/Software/Cryptic/segs exists and does NOT contain install_dir/path ?

**Comment by broxen** [comment]
Regedit has SEGS but does not contain install_dir/path.

Terminal output
```
[broxen@zelda out]$ cd /home/broxen/.wine32/dosdevices/c:/Program\ Files/CoX/
[broxen@zelda CoX]$ wine CoX.exe -project segs -fullscreen 0 -screen 1280 720
fixme:winediag:start_process Wine Staging 2.18 is a testing version containing experimental patches.
fixme:winediag:start_process Please mention your exact version when filing bug reports on winehq.org.
err:winediag:xrandr12_init_modes Broken NVIDIA RandR detected, falling back to RandR 1.0. Please consider using the Nouveau driver instead.
err:winediag:xrandr12_init_modes Broken NVIDIA RandR detected, falling back to RandR 1.0. Please consider using the Nouveau driver instead.
err:winediag:xrandr12_init_modes Broken NVIDIA RandR detected, falling back to RandR 1.0. Please consider using the Nouveau driver instead.
err:winediag:xrandr12_init_modes Broken NVIDIA RandR detected, falling back to RandR 1.0. Please consider using the Nouveau driver instead.
err:winediag:xrandr12_init_modes Broken NVIDIA RandR detected, falling back to RandR 1.0. Please consider using the Nouveau driver instead.
Running City Of Heroes
project segs
fullscreen 0
screen 1280 720
Caching relevant folders.. done                                       (0.00)
Loading texture headers..                                             (0.06)
CPU: 3500 Mhz / Memory: 2047 MBs / Video Card: Wine Display Adapter / Driver Version: Unknown Vendor /Available Memory: 2047 MBs
RENDER SETTINGS: VBOS 1, PARTICLE VBOS 0 BUMPMAPS 1, VERTSHADERS 1 
fixme:dbghelp:elf_search_auxv can't find symbol in module
fixme:dbghelp:MiniDumpWriteDump NIY MiniDumpWithDataSegs
err:ntdll:RtlpWaitForCriticalSection section 0x1685fc "?" wait timed out in thread 003f, blocked by 0000, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7be66c30 "dinput_main.c: dinput_hook_crit" wait timed out in thread 0009, blocked by 003f, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x1685fc "?" wait timed out in thread 003f, blocked by 0000, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7be66c30 "dinput_main.c: dinput_hook_crit" wait timed out in thread 0009, blocked by 003f, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x1685fc "?" wait timed out in thread 003f, blocked by 0000, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7be66c30 "dinput_main.c: dinput_hook_crit" wait timed out in thread 0009, blocked by 003f, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x1685fc "?" wait timed out in thread 003f, blocked by 0000, retrying (60 sec)
err:ntdll:RtlpWaitForCriticalSection section 0x7be66c30 "dinput_main.c: dinput_hook_crit" wait timed out in thread 0009, blocked by 003f, retrying (60 sec)
```
This is the 64bit wine prefix.

**Comment by nemerle** [comment]
What is interesting, this time it did not fail to load the tricks and other required files, so next step, still in the CoX directory
```bash
WINEPREFIX=/home/broxen/.wine32 wine CoX.exe -project segs -fullscreen 0 -screen 1280 720
```

**Comment by broxen** [comment]
This worked!

So I'm inferring from all of this that it's some bug in the way wine processes the commandline args when run from another folder?

**Comment by broxen** [comment]
Server window output, for reference.

```
main
(15987|140559856141440) Reading game data
(15987|140559856141440) Reading Costumes ... (15987|140559856141440) Couldn't load Costume!!
(15987|140559856141440) Reading Supergroup colors ... (15987|140559856141440) Couldn't load Supergroup colors!!
(15987|140559856141440) All game data read
(15987|140559856141440) Filling hashes .. Hashes filled
(15987|140559647942400) Connection closed
(15987|140559647942400) Connection closed
(15987|140559647942400) Connection closed
(15987|140559647942400) Connection closed
(15987|140559647942400) Connection closed
 Telling map server to expect a client with character EvilTest,1
Sending msg to client 0x7fd684003fa0
Received cookie confirm a00d741 - 0
Received window state 28 - 4
Received window state 28 - 1
Received window state 28 - 2
Received window state 28 - 4
Received window state 28 - 1
Received window state 28 - 2
Received window state 28 - 4
Received window state 28 - 1
Received window state 28 - 2
Received window state 11 - 4
Received window state 14 - 4
Received window state 15 - 4
Received window state 19 - 4
Received window state 20 - 4
Received window state 23 - 4
Received window state 24 - 4
Received window state 25 - 4
Received window state 27 - 4
Received window state 28 - 4
Received window state 29 - 4
Received window state 30 - 4
Received window state 31 - 4
Received window state 32 - 4
Received window state 33 - 4
Received window state 34 - 4
Pyr 0.000000 : 0.000000 
Pyr 0.000000 : 0.000000 
Received window state 24 - 4
Received window state 18 - 4
Received window state 17 - 4
Received window state 16 - 4
Received window state 8 - 4
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
Pyr 0.003068 : 0.000000 
Pyr 0.003068 : 0.000000 
Pyr 0.015340 : 0.000000 
Pyr 0.015340 : -0.003068 
Pyr 0.024544 : -0.003068 
Pyr 0.024544 : -0.006136 
Pyr 0.030680 : -0.006136 
Pyr 0.030680 : -0.015340 
Pyr 0.015340 : 0.000000 
Pyr 0.015340 : -0.003068 
Pyr 0.024544 : -0.003068 
Pyr 0.024544 : -0.006136 
Pyr 0.030680 : -0.006136 
Pyr 0.030680 : -0.015340 
Pyr 0.039884 : -0.015340 
Pyr 0.039884 : -0.024544 
Pyr 0.046019 : -0.024544 
Pyr 0.046019 : -0.042951 
Pyr 0.055223 : -0.042951 
Pyr 0.055223 : -0.098175 
Pyr 0.039884 : 0.000000 
Pyr 0.039884 : -0.024544 
Pyr 0.046019 : -0.024544 
Pyr 0.046019 : -0.042951 
Pyr 0.055223 : -0.042951 
Pyr 0.055223 : -0.098175 
Pyr 0.067495 : -0.098175 
Pyr 0.067495 : -0.159534 
Pyr 0.076699 : -0.159534 
Pyr 0.076699 : -0.217825 
Pyr 0.085903 : -0.217825 
Pyr 0.085903 : -0.269981 
Pyr 0.067495 : 0.000000 
Pyr 0.067495 : -0.159534 
Pyr 0.076699 : -0.159534 
Pyr 0.076699 : -0.217825 
Pyr 0.085903 : -0.217825 
Pyr 0.085903 : -0.269981 
Pyr 0.098175 : -0.269981 
Pyr 0.098175 : -0.346680 
Pyr 0.101243 : -0.346680 
Pyr 0.101243 : -0.414175 
Pyr 0.113515 : -0.414175 
Pyr 0.113515 : -0.524621 
Pyr 0.122718 : 0.000000 
Pyr 0.122718 : -0.714835 
Pyr 0.116583 : -0.714835 
Pyr 0.116583 : -0.754719 
Pyr 0.107379 : -0.754719 
Pyr 0.107379 : -0.880505 
Pyr 0.122718 : 0.000000 
Pyr 0.122718 : -0.714835 
Pyr 0.116583 : -0.714835 
Pyr 0.116583 : -0.754719 
Pyr 0.107379 : -0.754719 
Pyr 0.107379 : -0.880505 
Pyr 0.116583 : -0.880505 
Pyr 0.116583 : -1.030835 
Pyr 0.138058 : -1.030835 
Pyr 0.138058 : -1.168893 
Pyr 0.147262 : -1.168893 
Pyr 0.147262 : -1.257864 
Pyr 0.116583 : 0.000000 
Pyr 0.116583 : -1.030835 
Pyr 0.138058 : -1.030835 
Pyr 0.138058 : -1.168893 
Pyr 0.147262 : -1.168893 
Pyr 0.147262 : -1.257864 
Pyr 0.159534 : -1.257864 
Pyr 0.159534 : -1.313088 
Pyr 0.190214 : -1.313088 
Pyr 0.190214 : -1.457282 
Pyr 0.177942 : -1.457282 
Pyr 0.177942 : -1.576932 
Pyr 0.187146 : 0.000000 
Pyr 0.187146 : -1.567728 
Pyr 0.193282 : -1.567728 
Pyr 0.193282 : -1.549321 
Pyr 0.187146 : 0.000000 
Pyr 0.187146 : -1.567728 
Pyr 0.193282 : -1.567728 
Pyr 0.193282 : -1.549321 
Pyr 0.187146 : 0.000000 
Pyr 0.162602 : 0.000000 
Pyr 0.162602 : -1.567728 
Pyr 0.119650 : -1.567728 
Pyr 0.119650 : -1.601476 
Pyr 0.187146 : 0.000000 
Pyr 0.162602 : 0.000000 
Pyr 0.162602 : -1.567728 
Pyr 0.119650 : -1.567728 
Pyr 0.119650 : -1.601476 
Pyr 0.070563 : -1.601476 
Pyr 0.070563 : -1.653631 
Pyr 0.021476 : -1.653631 
Pyr 0.021476 : -1.699651 
Pyr -0.036816 : -1.699651 
Pyr -0.036816 : -1.751806 
Pyr 0.070563 : 0.000000 
Pyr 0.070563 : -1.653631 
Pyr 0.021476 : -1.653631 
Pyr 0.021476 : -1.699651 
Pyr -0.036816 : -1.699651 
Pyr -0.036816 : -1.751806 
Pyr -0.085903 : -1.751806 
Pyr -0.085903 : -1.797825 
Pyr -0.113515 : -1.797825 
Pyr -0.113515 : -1.816233 
Pyr -0.171806 : -1.816233 
Pyr -0.171806 : -1.849981 
Pyr -0.085903 : 0.000000 
Pyr -0.085903 : -1.797825 
Pyr -0.113515 : -1.797825 
Pyr -0.113515 : -1.816233 
Pyr -0.171806 : -1.816233 
Pyr -0.171806 : -1.849981 
Pyr -0.217825 : -1.849981 
Pyr -0.217825 : -1.871457 
Pyr -0.276117 : -1.871457 
Pyr -0.276117 : -1.902136 
Pyr -0.358952 : -1.902136 
Pyr -0.358952 : -1.975767 
Pyr -0.389631 : 0.000000 
Pyr -0.389631 : -2.006447 
Pyr -0.450990 : -2.006447 
Pyr -0.450990 : -2.098486 
Pyr -0.490874 : -2.098486 
Pyr -0.490874 : -2.181321 
Pyr -0.389631 : 0.000000 
Pyr -0.389631 : -2.006447 
Pyr -0.450990 : -2.006447 
Pyr -0.450990 : -2.098486 
Pyr -0.490874 : -2.098486 
Pyr -0.490874 : -2.181321 
Pyr -0.527689 : -2.181321 
Pyr -0.527689 : -2.285631 
Pyr -0.579845 : -2.285631 
Pyr -0.579845 : -2.432894 
Pyr -0.598253 : -2.432894 
Pyr -0.598253 : -2.524932 
Pyr -0.527689 : 0.000000 
Pyr -0.527689 : -2.285631 
Pyr -0.579845 : -2.285631 
Pyr -0.579845 : -2.432894 
Pyr -0.598253 : -2.432894 
Pyr -0.598253 : -2.524932 
Pyr -0.607456 : -2.524932 
Pyr -0.607456 : -2.561748 
Pyr -0.601320 : -2.561748 
Pyr -0.601320 : -2.687534 
Pyr -0.585981 : 0.000000 
Pyr -0.585981 : -3.083301 
Pyr -0.579845 : -3.083301 
Pyr -0.579845 : -3.120117 
Pyr -0.573709 : -3.120117 
Pyr -0.573709 : 3.009670 
Pyr -0.543029 : 3.009670 
Pyr -0.543029 : 2.877748 
Pyr -0.585981 : 0.000000 
Pyr -0.585981 : -3.083301 
Pyr -0.579845 : -3.083301 
Pyr -0.579845 : -3.120117 
Pyr -0.573709 : -3.120117 
Pyr -0.573709 : 3.009670 
Pyr -0.543029 : 3.009670 
Pyr -0.543029 : 2.877748 
Pyr -0.500078 : 2.877748 
Pyr -0.500078 : 2.755029 
Pyr -0.450990 : 2.755029 
Pyr -0.450990 : 2.641515 
Pyr -0.414175 : 0.000000 
Pyr -0.414175 : 2.675262 
Pyr -0.404971 : 2.675262 
Pyr -0.404971 : 2.699806 
Pyr -0.414175 : 0.000000 
Pyr -0.414175 : 2.675262 
Pyr -0.404971 : 2.675262 
Pyr -0.404971 : 2.699806 
Pyr -0.408039 : 0.000000 
Pyr -0.423379 : 0.000000 
Pyr -0.435651 : 0.000000 
Pyr -0.435651 : 2.709010 
Pyr -0.447922 : 2.709010 
Pyr -0.447922 : 2.745826 
Pyr -0.423379 : 0.000000 
Pyr -0.435651 : 0.000000 
Pyr -0.435651 : 2.709010 
Pyr -0.447922 : 2.709010 
Pyr -0.447922 : 2.745826 
Pyr -0.460194 : 2.745826 
Pyr -0.460194 : 2.776505 
Pyr -0.432583 : 2.776505 
Pyr -0.432583 : 2.804117 
Pyr -0.340544 : 2.804117 
Pyr -0.340544 : 2.856272 
Pyr -0.309864 : 0.000000 
Pyr -0.309864 : 2.865476 
Pyr -0.303728 : 2.865476 
Pyr -0.303728 : 2.868544 
Pyr -0.288388 : 2.868544 
Pyr -0.288388 : 2.871612 
Pyr -0.309864 : 0.000000 
Pyr -0.309864 : 2.865476 
Pyr -0.303728 : 2.865476 
Pyr -0.303728 : 2.868544 
Pyr -0.288388 : 2.868544 
Pyr -0.288388 : 2.871612 
Pyr -0.285320 : 2.871612 
Pyr -0.294524 : 2.871612 
Pyr -0.294524 : 2.877748 
Pyr -0.297592 : 2.877748 
Pyr -0.297592 : 2.880816 
Pyr -0.285320 : 0.000000 
Pyr -0.294524 : 0.000000 
Pyr -0.294524 : 2.877748 
Pyr -0.297592 : 2.877748 
Pyr -0.297592 : 2.880816 
Pyr -0.291456 : 2.880816 
Pyr -0.291456 : 2.883884 
Pyr -0.285320 : 2.883884 
Pyr -0.285320 : 2.886952 
Pyr -0.266913 : 2.886952 
Pyr -0.266913 : 2.902292 
Pyr -0.211689 : 0.000000 
Pyr -0.211689 : 2.948311 
Pyr -0.193282 : 2.948311 
Pyr -0.193282 : 2.994330 
Pyr -0.181010 : 2.994330 
Pyr -0.181010 : 3.021942 
Pyr -0.165670 : 3.021942 
Pyr -0.165670 : 3.052622 
Pyr -0.211689 : 0.000000 
Pyr -0.211689 : 2.948311 
Pyr -0.193282 : 2.948311 
Pyr -0.193282 : 2.994330 
Pyr -0.181010 : 2.994330 
Pyr -0.181010 : 3.021942 
Pyr -0.165670 : 3.021942 
Pyr -0.165670 : 3.052622 
Pyr -0.147262 : 3.052622 
Pyr -0.147262 : 3.080233 
Pyr -0.122718 : 3.080233 
Pyr -0.122718 : 3.104777 
Pyr -0.079767 : 0.000000 
Pyr -0.079767 : 3.135457 
Pyr -0.052155 : 3.135457 
Pyr -0.052155 : -3.126253 
Pyr -0.042951 : -3.126253 
Pyr -0.042951 : -3.120117 
Pyr -0.079767 : 0.000000 
Pyr -0.079767 : 3.135457 
Pyr -0.052155 : 3.135457 
Pyr -0.052155 : -3.126253 
Pyr -0.042951 : -3.126253 
Pyr -0.042951 : -3.120117 
Pyr -0.036816 : -3.120117 
Pyr -0.036816 : -3.117049 
Pyr -0.021476 : -3.117049 
Pyr -0.021476 : -3.107845 
Pyr -0.009204 : 0.000000 
Pyr -0.009204 : -3.104777 
Pyr -0.006136 : -3.104777 
Pyr -0.009204 : 0.000000 
Pyr -0.009204 : -3.104777 
Pyr -0.006136 : -3.104777 
Pyr -0.012272 : -3.104777 
Pyr -0.012272 : -3.110913 
Pyr -0.012272 : -3.113981 
Pyr -0.012272 : 0.000000 
Pyr -0.012272 : -3.110913 
Pyr -0.012272 : -3.113981 
pressed
pressed
Unhandled event type 116
Unhandled event type 116
Unhandled event type 116
pressed
pressed
pressed
pressed
Unhandled event type 116
pressed
pressed
Unhandled event type 116
pressed
pressed
Unhandled event type 116
Unhandled event type 116
pressed
pressed
```

**Comment by nemerle** [comment]
Client is likely not finding it's install dir so it can only be ran from it's directory, so we might actually WANT the install dir in the registry
as for costumes and colors you'll need to run:
```bash
cd SERVER_DIR
./piggtool -x PATH_TO_COH_INSTALL/piggs/bin.pigg
```

**Comment by broxen** [comment]
Can confirm that extracting the bin files to the data directory resolved the costume and colors issue.

I'm still showing up as semi-transparent? Also, let me know if I need to add anything to the registry RE: install_dir

![screenshot from 2017-10-21 00-39-35](https://user-images.githubusercontent.com/10159505/31849540-fc12dd4e-b5f8-11e7-971a-575e21ae0146.png)

**Comment by nemerle** [comment]
being semi-transparent is likely some wrong value is sent with Entity definition.
The registry value should be named ```Installation Directory``` and be under Cryptic/segs key.

**Comment by broxen** [comment]
Added Installation Directory, but still cannot load from another folder. Same error messages as before. That said, further digging revealed that the issue was my misuse of the wine command line. So I think we're fine here and can _close this issue._

You need to use `wine start` if you specify a full path to the .exe, because that allows Wine to set the working directory for the program if it needs it.

More info on [wiki.winehq.org](https://wiki.winehq.org/Wine_User%27s_Guide#How_to_run_Windows_programs_from_the_command_line)

Long story short, I was using wine wrong.

**Comment by broxen** [comment]
The correct command (for the curious):
`env WINEPREFIX="$HOME/.wine32" wine start /unix $HOME/.wine32/dosdevices/c:/Program\ Files/CoX/CoX.exe -project segs -fullscreen 0 -screen 1720 900`
